### DISCUSSION TITLE
(mini.comment) Option 'commentstring' is empty.

### DISCUSSION THREAD

**Discussion by tommyalatalo** [LazyVim/LazyVim#654] [Q&A]
Has something changed with neovim 0.9.0, lazyvim or thhe mini.comment plugin recently to cause the error `(mini.comment) Option 'commentstring' is empty.` when trying to apply a line comment with `gcc`?

This just started happening now, running neovim 0.9.0 and lazyvim synced with the latest mini.comment:

```
NVIM v0.9.0
Build type: Release
LuaJIT 2.1.0-beta3

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "/usr/share/nvim"

Run :checkhealth for more info
```

**Comment by folke** [comment]
For what filetype? I don't have any issue with this

**Reply by tommyalatalo** [reply]
I'm seeing this for `.hcl` files, but only on one of my hosts where I use neovim, that's odd. Maybe I should just nuke the .local/share/nvim folder in case there's some kind of weird conflict happening...?

**Reply by scottbenjamin** [reply]
I'm also seeing this as of recently, also with `.hcl` files.

**Reply by tommyalatalo** [reply]
Yeah this is still an issue with `.hcl` files on my work machine, which oddly is not the case on my own machine.
Is this something that should be reported to mini.comment?

**Reply by folke** [reply]
Then it seems that hcl files dont have a commenstring set? That has nothing to do with mini.comment

**Reply by tommyalatalo** [reply]
That's odd, this used to work until some update broke it.
I'm not familiar with commentstring as a concept, where is that usually defined?

**Reply by folke** [reply]
Check `:h 'commentstring'`

Maybe a Neovim update broke it for hcl files

**Reply by jonaustin** [reply]
For anyone that comes across this for terraform files; I had to set this manually:

```
vim.api.nvim_create_autocmd("FileType", {
  pattern = "terraform",
  callback = function()
    vim.bo.commentstring = "# %s"
  end
})
```

I haven't tried it, but alternatively looks like: https://github.com/JoosepAlviste/nvim-ts-context-commentstring
should handle it automatically.

**Reply by kumkee** [reply]
> vim.bo.commentstring = "# %s"

This can also be done by adding
``` 
vim.bo.commentstring = "# %s"
```
to `terraform.lua` in the `ftplugin` folder.
